pkg/segment/query/processor/dataprocessor.go

```diff
@@ -90,6 +90,8 @@ func (dp *DataProcessor) Fetch() (*iqr.IQR, error) {
 			gotEOF = true
 		} else if err != nil {
 			return nil, utils.TeeErrorf("DP.Fetch: failed to process input: %v", err)
+		} else if output == nil && dp.isBottleneckCmd {
+			gotEOF = false
```

Comment:
I'm concerned about this change; it feels more like a hack than a proper solution. Why do we need this? Also I think this will fail (i.e., cause an infinite loop) if we get to a situation where the bottleneck command _should_ return no output (e.g., a sort where no data was passed in)

Reply:
But wouldn't the bottleneck command should return `io.EOF` at some point? Then it does not matter what the output is, the `EOF` will be considered and it will break out of loop.

Also I am changing this part a bit.